### PR TITLE
Allow variable number of servers in `run-ui`

### LINKED ISSUE #2512: `run-ui` does not respect fed size argument
1. `run-ui.sh` has tails [hard coded for 0 and 1](https://github.com/fedimint/fedimint/blob/master/scripts/run-ui.sh#L10-L13), so it doesn't tail more than 2 federations' logs
2. `run_ui` iters over a [hard coded 2 fedimintds](https://github.com/fedimint/fedimint/blob/master/devimint/src/main.rs#L897), so only 2 start

### PR DESCRIPTION
**Pull request by wbobeirne**
Closes #2512

### CODE REVIEW

File: scripts/run-ui.sh [hunk 1]
```diff
@@ -4,12 +4,15 @@ export FM_BITCOIN_NETWORK="regtest"
 export FM_UI_KIND=${1:-"old"}
 export FM_FED_SIZE=${2:-2}
 export FM_FED_NAME=${3:-"Cypherpunk Federation"}
+export RUST_BACKTRACE="full"
```
Comment (Maan2003) [inline scripts/run-ui.sh:7]: why full? RUST_BACKTRACE=1 has more readable backtrace.

File: devimint/src/main.rs [hunk 2]
```diff
@@ -974,8 +977,9 @@ async fn main() -> Result<()> {
             task_group.make_handle().make_shutdown_rx().await.await?;
         }
         Cmd::RunUi(kind) => {
+            let fed_size = args.common.fed_size;
             let (process_mgr, task_group) = setup(args.common).await?;
-            run_ui(&process_mgr, &task_group, &kind).await?
+            run_ui(&process_mgr, &task_group, &kind, fed_size).await?
```
Comment (Maan2003) [inline devimint/src/main.rs:982]: not blocking, 
but you can also use `process_mgr.globals.FM_FED_SIZE` this is set in `setup` function `var::Global::new(..., 4)`

File: scripts/run-ui.sh (hunk 1 above)
Comment (wbobeirne) [reply scripts/run-ui.sh:7]: Whoops, left this in from debugging a panic I ran into while running this. I'll remove.

File: devimint/src/main.rs (hunk 2 above)
Comment (wbobeirne) [reply devimint/src/main.rs:982]: Got it. New to the Rust side so appreciate the pointers. I just looked for somewhere else where we were accessing an env var and tried to copy the pattern.

Down in globals, it looks to me like it's hard coded to 4? Is there something I'm missing where it's reading in the env var? Or are you recommending I update that `4` to be `arg.fed_size` instead?

Comment (Maan2003) [reply devimint/src/main.rs:982]: yes, that 4 should change to `arg.fed_size`

Comment (Maan2003) [reply devimint/src/main.rs:982]: you can also try replacing other 4s with globals.FM_FED_SIZE  (`rg '\b4\b' devimint/src`)
But can also be done in a followup.

File: scripts/run-ui.sh (hunk 1 above)
Comment (EthnTuttle) [reply scripts/run-ui.sh:7]:
```suggestion
```
Just commenting so I get alerts when it's merged.

File: devimint/src/main.rs (hunk 2 above)
Comment (wbobeirne) [reply devimint/src/main.rs]: Went ahead and did it here. Thanks for the note!

**Review by Maan2003 — APPROVED** [review]

**Pull request merged**